### PR TITLE
create a hubspot connector

### PR DESCRIPTION
**Pull request by nlp8899**

### CODE REVIEW

File: web/src/app/admin/layout.tsx [hunk 1]
```diff
@@ -192,6 +193,15 @@ export default async function AdminLayout({
                   ),
                   link: "/admin/connectors/file",
                 },
+                {
+                  name: (
+                    <div className="flex">
+                      <BookstackIcon size={16} />
```
Comment (yuhongsun96) [inline web/src/app/admin/layout.tsx]: Probably want to use the HubSpot icon here

File: backend/danswer/connectors/hubspot/connector.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                doc_batch = []
+
+        if doc_batch:
+            yield doc_batch
+
+    def load_from_state(self) -> GenerateDocumentsOutput:
+        return self._process_tickets()
+
+    def poll_source(self, start: int, end: int) -> GenerateDocumentsOutput:
+        # Currently, we're loading all tickets. Adjust this method if HubSpot provides a way to load tickets by date range.
```
Comment (yuhongsun96) [inline backend/danswer/connectors/hubspot/connector.py]: If we can get the ticket last update time, we can throw out the ones not in the range, even if there's no way to provide a time filter upfront. The largest slowdown in indexing time comes from the NLP models embedding documents so even if we fetch the tickets and throw them out before passing them to the model, there's a large efficiency gain!

**Comment by yuhongsun96** [comment]
Nice! Thanks for the contribution. Added two small comments in the code, would love to get your thoughts. 

Also since I'm not totally familiar with HubSpot, I wanted to know the thinking behind pulling "tickets" in particular. I read that HubSpot has knowledge bases with wiki features and things like call transcripts etc.

One other thing, if we could get a docs page for HubSpot that would be really sweet to have as well! I can take it up if you'd like but I think you'd do a better job being more familiar with HubSpot!

File: web/src/app/admin/layout.tsx (hunk 1 above)
Comment (nlp8899) [reply web/src/app/admin/layout.tsx]: Yes, I will update this. Thank you for your catching.

File: backend/danswer/connectors/hubspot/connector.py (hunk 2 above)
Comment (nlp8899) [reply backend/danswer/connectors/hubspot/connector.py]: Okay, I will update this part after discussing with Roberto.

**Comment by yuhongsun96** [comment]
Great work, thanks for adding the polling support!

**Comment by nlp8899** [comment]
> Great work, thanks for adding the polling support!

Thank you

**Pull request merged**